src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -104,17 +111,66 @@ private void UpdateNextRequestUri(ResponseHeaders headers)
             switch (_headerSource)
             {
                 case HeaderSource.OperationLocation when headers.TryGetValue("Operation-Location", out string? operationLocation):
-                    _nextRequestUri = operationLocation;
+                    _nextRequestUri = AppendOrReplaceApiVersion(operationLocation, _apiVersion);
                     return;
                 case HeaderSource.AzureAsyncOperation when headers.TryGetValue("Azure-AsyncOperation", out string? azureAsyncOperation):
-                    _nextRequestUri = azureAsyncOperation;
+                    _nextRequestUri = AppendOrReplaceApiVersion(azureAsyncOperation, _apiVersion);
                     return;
                 case HeaderSource.Location when hasLocation:
-                    _nextRequestUri = location!;
+                    _nextRequestUri = AppendOrReplaceApiVersion(location!, _apiVersion);
                     return;
             }
         }
 
+        internal static string AppendOrReplaceApiVersion(string uri, string? apiVersion)
+        {
+            if (apiVersion != null)
+            {
+                var uriSpan = uri.AsSpan();
+                var apiVersionParamSpan = ApiVersionParam.AsSpan();
+                var apiVersionIndex = uriSpan.IndexOf(apiVersionParamSpan);
+                if (apiVersionIndex == -1)
+                {
+                    var concatSymbol = uriSpan.IndexOf('?') > -1 ? "&" : "?";
+                    return $"{uri}{concatSymbol}api-version={apiVersion}";
+                }
+                else
+                {
+                    var lengthToEqualSignAfterApiVersionParam = apiVersionIndex + ApiVersionParam.Length + 1;
```

Comment:
Nit: while incredibly unlikely, params don't necessary have to be followed by `=`. Might want to actually make sure. Again, though, very unlikely. Perhaps causing a service error in such a case is actually desirable to find issues.

Reply:
Such case is also handled and I've added corresponding tests.